dialogue-target/src/main/java/com/palantir/dialogue/ExceptionDeserializerArgs.java

```diff
@@ -128,6 +130,22 @@ public TypeMarker<T> returnType() {
         return errorNameToExceptionTypeMarkers;
     }
 
-    public record ErrorExceptionPair<U extends AbstractSerializableError<?>, V extends RemoteException>(
-            TypeMarker<U> errorType, TypeMarker<V> exceptionType) {}
+    /**
+     * This record is used in the implementation of Dialogue's error decoder. This should not be used by clients. It
+     * is public to allow usage in the implementation in the com.palantir.conjure.java.dialogue.serde package.
+     **/
+    public record ErrorExceptionPair<
+            U extends AbstractSerializableError<?>, V extends RemoteException & SerializableErrorProvider<?>>(
+            TypeMarker<U> errorType, TypeMarker<V> exceptionType) {
+        @SuppressWarnings("unchecked")
+        public static <U extends AbstractSerializableError<?>, V extends RemoteException & SerializableErrorProvider<?>>
+                V getExceptionFromSerializableError(U error, TypeMarker<V> exceptionType, int code)
+                        throws InvocationTargetException, InstantiationException, IllegalAccessException,
+                                NoSuchMethodException, ClassNotFoundException {
+            Class<V> exceptionClass =
+                    (Class<V>) Class.forName(exceptionType.getType().getTypeName());
+            Constructor<V> exceptionConstructor = exceptionClass.getConstructor(error.getClass(), int.class);
```

Comment:
vnit: We could add a comment explaining that 
```suggestion
            // RemoteExceptions constructible from a serializable error MUST have a constructor that takes such an error as well as the http error code
            Constructor<V> exceptionConstructor = exceptionClass.getConstructor(error.getClass(), int.class);
```
and maybe add a try-catch that throws a more explicit exception saying something along the lines of "could not find a valid constructor for RemoteException subclass to construct from provided serializable error" with both class names as argument, which will make it easier to debug if this ever happens

That would also let us encapsulate the `NoSuchMethodException` and `ClassNotFoundException` cases here 🤷‍♂️

Reply:
Added. Also I previously added a comment to the method in the builder.